### PR TITLE
Introduce transcription protocol

### PR DESCRIPTION
**Pull request by keepingitneil**

### CODE REVIEW

**Review by davidzhao — APPROVED** [review]
lgtm!

thanks for fixing the CI too

File: livekit-rtc/livekit/rtc/participant.py [hunk 1]
```diff
@@ -117,6 +131,42 @@ async def publish_data(
         if cb.publish_data.error:
             raise PublishDataError(cb.publish_data.error)
 
+    async def publish_transcription(
+        self,
+        participant_identity: str,
```
Comment (theomonnom) [inline livekit-rtc/livekit/rtc/participant.py:136]: I would be more comfortable creating a Transcription structure here.
On the agents API, we will need to return a Transcription, and I don't think this should be defined inside agents.

publish_transcription accepting a "Transcription" seems OK to me

File: livekit-rtc/livekit/rtc/transcription.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from dataclasses import dataclass
+
+
```
Comment (theomonnom) [inline livekit-rtc/livekit/rtc/transcription.py:3]: Add it here?

**Review by theomonnom — CHANGES_REQUESTED** [review]
I'm also wondering if start_time and end_time should be float on the Python side.
To respect the time.time() format.

Maybe we could automatically do the conversion?

**Comment by keepingitneil** [comment]
@theomonnom re: float. I think we should keep it int because the clock being used will be the frame clock. Although I'm wondering if we should specify units in the transcription protocol now: `start_time_us` ?

**Review by theomonnom — APPROVED** [review]

**Comment by theomonnom** [comment]
let's mb wait for a release tho?

**Pull request merged**